chain/actions/src/redeem.rs

```diff
@@ -155,12 +161,18 @@ where
             .domain_separator(DomainSeparator::Channel)
             .ok_or(InvalidState("missing channel dst".into()))?;
 
-        let mut redeem_stream = self
+        let selector_id = selector.to_string();
+
+        // Collect here, so we don't hold-up the stream open for too long
+        let redeem_stream = self
             .db
             .update_ticket_states_and_fetch(selector, AcknowledgedTicketStatus::BeingRedeemed)
-            .await?;
+            .await?
+            .collect::<Vec<_>>()
```

Comment:
At this point does it make sense to use db streams internally still?

Reply:
If this change works, then no. They seem to be not suitable for SQLite backends.